instrumentation/influxdb-2.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/influxdb/v2_4/InfluxDbClientTest.java

```diff
@@ -0,0 +1,311 @@
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import org.influxdb.InfluxDB;
+import org.influxdb.InfluxDBFactory;
+import org.influxdb.dto.BatchPoints;
+import org.influxdb.dto.Point;
+import org.influxdb.dto.Query;
+import org.influxdb.dto.QueryResult;
+import org.junit.Assert;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.testcontainers.containers.GenericContainer;
+
+// ignore @Deprecated annotation warning.
+@SuppressWarnings("deprecation")
+@TestInstance(Lifecycle.PER_CLASS)
+class InfluxDbClientTest {
+
+  @RegisterExtension
+  private static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  private static final GenericContainer<?> influxDbServer =
+      new GenericContainer<>("influxdb:1.8.10-alpine").withExposedPorts(8086);
+
+  private static InfluxDB influxDb;
+
+  private static final String databaseName = "mydb";
+
+  private static String serverURL;
+
+  private static int port;
+
+  @BeforeAll
+  void setup() {
+    influxDbServer.start();
+    port = influxDbServer.getMappedPort(8086);
+    String host = influxDbServer.getHost();
+    serverURL = "http://" + host + ":" + port + "/";
+    String username = "root";
+    String password = "root";
+    influxDb = InfluxDBFactory.connect(serverURL, username, password);
+    influxDb.createDatabase(databaseName);
+  }
+
+  @AfterAll
+  void cleanup() {
+    influxDb.deleteDatabase(databaseName);
+    influxDb.close();
+  }
+
+  @BeforeEach
+  void reset() {
+    testing.clearData();
+  }
+
+  @Test
+  void testQueryAndModifyWithOneArgument() {
+    String dbName = databaseName + System.currentTimeMillis();
+    influxDb.createDatabase(dbName);
+    BatchPoints batchPoints =
+        BatchPoints.database(dbName).tag("async", "true").retentionPolicy("autogen").build();
+    Point point1 =
+        Point.measurement("cpu")
+            .tag("atag", "test")
+            .addField("idle", 90L)
+            .addField("usertime", 9L)
+            .addField("system", 1L)
+            .build();
+    Point point2 =
+        Point.measurement("disk")
+            .tag("atag", "test")
+            .addField("used", 80L)
+            .addField("free", 1L)
+            .build();
+    batchPoints.point(point1);
+    batchPoints.point(point2);
+    influxDb.write(batchPoints);
+    Query query = new Query("SELECT * FROM cpu GROUP BY *", dbName);
+    QueryResult result = influxDb.query(query);
+    Assert.assertFalse(result.getResults().get(0).getSeries().get(0).getTags().isEmpty());
```

Comment:
usually we use assertJ `assertThat(result.getResults().get(0).getSeries().get(0).getTags().isEmpty()).isFalse();` or better `assertThat(result.getResults().get(0).getSeries().get(0).getTags()).isEmpty();`